tests/phpunit/integration/Core/User_Surveys/REST_User_Surveys_ControllerTest.php

```diff
@@ -148,4 +301,28 @@ public function test_survey_timeout() {
 		);
 	}
 
+	public function test_survey() {
+		remove_all_filters( 'googlesitekit_rest_routes' );
+		$this->controller->register();
+		$this->register_rest_routes();
+
+		$survey = array(
+			'survey_id' => 'test_survey',
+			'session'   => array(
+				'session_id'    => 'test_session_id',
+				'session_token' => 'test_session_token',
+			),
+			'payload'   => array(
+				'questions' => array(),
+			),
+		);
+
+		$this->queue->enqueue( $survey );
```

Comment:
Let's test this with more than one survey in the queue to ensure that it's returning the right one.

Reply:
Updated.